packages/wallet/src/cip30.ts

```diff
@@ -92,9 +108,35 @@ export const createWalletApi = (
       throw new ApiError(500, 'Nope');
     }
   },
-  getCollateral: async () => {
-    logger.warn('getCollateral is not implemented');
-    return null;
+  getCollateral: async ({ amount = cslToCbor(MAX_COLLATERAL_AMOUNT) }: { amount?: Cbor } = {}): Promise<
+    Cbor[] | null
+  > => {
+    logger.debug('getting collateral');
+    const wallet = await walletReady;
+    let unspendables = (await firstValueFrom(wallet.utxo.unspendable$)).sort(compareUtxos);
+    if (amount) {
+      try {
+        const filterAmount = CSL.BigNum.from_bytes(Buffer.from(amount, 'hex'));
+        if (filterAmount.compare(MAX_COLLATERAL_AMOUNT) > 0) throw new ApiError(400, 'given amount is too big');
+        const utxos = [];
+        let totalCoins = CSL.BigNum.from_str('0');
+        for (const utxo of unspendables) {
+          const coin = CSL.BigNum.from_str(utxo[1].value.coins.toString());
+          totalCoins = totalCoins.checked_add(coin);
+          utxos.push(utxo);
+          if (totalCoins.compare(filterAmount) !== -1) break;
+        }
+        if (totalCoins.compare(filterAmount) === -1) throw new ApiError(400, 'not enough coins in unspendable UTxOs');
+        unspendables = utxos;
+      } catch (error) {
+        logger.error(error);
+        if (error instanceof ApiError) {
+          throw error;
+        }
+        throw new ApiError(500, 'Nope');
```

Comment:
```suggestion
        throw new ApiError(500, 'Nope'); // We can define more meaningful error message such as 'Error occurred during validating given amount of collateral' 
```

Reply:
I wouldn't say it is being use to validate the amount. I think it is used to handle unexpected errors. So, message would be something like 'Unexpected error occured'. It should be changed in other methods to be constant.